### ISSUE TITLE
Add a mechanism for disabling @mentions on a page

### ISSUE THREAD

**Issue by bkeepers** [jekyll/jekyll-mentions#39]
I'm following the [Jekyll search using lunr.js](http://jekyll.tips/jekyll-casts/jekyll-search-using-lunr-js/) guide, and one step involves outputting JSON on the page:

```html
…
<script>
  window.store = {
    {% for post in site.posts %}
      "{{ post.url | slugify }}": {
        "title": "{{ post.title | xml_escape }}",
        "author": "{{ post.author | xml_escape }}",
        "category": "{{ post.category | xml_escape }}",
        "content": {{ post.content | strip_html | strip_newlines | jsonify }},
        "url": "{{ post.url | xml_escape }}"
      }
      {% unless forloop.last %},{% endunless %}
    {% endfor %}
  };
</script>
…
```

This works fine, except that the jekyll-mentions plugin still finds @mentions in `content` and autolinks them, which then breaks the json.

For this specific use case, one solution is to move just the json to a separate file called `content.json`, since jekyll-mentions only operates on [html pages](https://github.com/jekyll/jekyll-mentions/blob/851d5a317a776f3fc34dd5cc1153adbd03c3709d/lib/jekyll-mentions.rb#L83).

Are there other uses cases though where it might make sense to allow disabling @mentions per-page?

**Comment by benbalter** [comment]
> This works fine, except that the jekyll-mentions plugin still finds @mentions in content and autolinks them, which then breaks the json.

Is it linking the mention once it's been jsonified, or the content when it renders the underlying post?

> For this specific use case, one solution is to move just the json to a separate file called content.json, since jekyll-mentions only operates on html pages.

Exactly what I was going to suggest.

**Comment by bradjc** [comment]
> Are there other uses cases though where it might make sense to allow disabling @mentions per-page?

Yes and no. The case is if you include CSS in a page and it has an `@media` attribute. However, disabling in this case _absolutely_ makes sense.

**Comment by mbirth** [comment]
I'd prefer a way to prevent single `@…` statements from being linked, e.g. by escaping the `@` like in `\@something`. This is how it works for other markdown, too.

**Comment by parkr** [comment]
I'm wholly 👍 to the idea of a front-matter option like `jekyll-mentions: disable` or `jekyll-mentions: false` or `disable_plugins: [jekyll-mentions]` on a per-page/per-document basis.

**Comment by emma-sax4** [comment]
Also, would be super nice if you could override the `base_url` for a single file, such as
```
jekyll-mentions: https://twitter.com
```
in your `_config.yml`, but 
```
jekyll-mentions: https://github.com
```
for a different file to use that base_url.

**Comment by ashmaroli** [comment]
@bkeepers I would like to inform you that this feature has landed on the `master`!! :tada: :sparkles:
If you're still using this plugin, do let us know if it works for your use-case. Thanks.
